superset/assets/javascripts/explore/components/controls/FixedOrMetricControl.jsx

```diff
@@ -0,0 +1,118 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Label, Popover, OverlayTrigger } from 'react-bootstrap';
+
+import controls from '../../stores/controls';
+import TextControl from './TextControl';
+import SelectControl from './SelectControl';
+import ControlHeader from '../ControlHeader';
+import PopoverSection from '../../../components/PopoverSection';
+
+const propTypes = {
+  onChange: PropTypes.func,
+  value: PropTypes.object,
+  isFloat: PropTypes.bool,
+  datasource: PropTypes.object,
+  default: PropTypes.object,
+};
+
+const defaultProps = {
+  onChange: () => {},
+  default: { type: 'fix', value: 5 },
+};
+
+export default class FixedOrMetricControl extends React.Component {
+  constructor(props) {
+    super(props);
+    this.onChange = this.onChange.bind(this);
+    const type = props.value ? props.value.type : props.default.type || 'fix';
```

Comment:
'fix' should probably be a global constant that you reference everywhere. e.g. declare `VALUE_TYPE_FIX = 'fix'` and use that everywhere. In case you ever want to refactor, this will make it a lot easier.

Reply:
I think adding parenthesis will make this line easier to read:
```
const type = props.value ? props.value.type : (props.default.type || 'fix');
vs
const type = (props.value ? props.value.type : props.default.type) || 'fix';
```
I always get confused about the order of operations in these cases, but maybe thats just me.
another option is to pull out `if (props.value)`

Reply:
yes like an enum, I'm going with
```
const controlTypes = {
  fixed: 'fix',
  metric: 'metric',
};
```